src/pages/references/node-configuration.md

```diff
@@ -0,0 +1,489 @@
+
+Example:
+
+```toml
+seed = "replace-with-your-private-key"
+```
+
+#### local_peer_seed (optional)
+
+The private key to use for mining.
+
+[See this page for information on how to generate a private key.](https://docs.blockstack.org/mining)
+
+Example:
+
+```toml
+local_peer_seed = "replace-with-your-private-key"
+```
+
+#### miner (optional)
+
+Determines whether the stacks-node is running a follower (`false`) or a miner (`true`).
+
+[See this page for information on how to run a miner.](https://docs.blockstack.org/mining)
+
+Example:
+
+```toml
+miner = true
+```
+
+#### mine_microblocks (optional)
+
+Determines whether the stacks-node will mine microblocks. Will only take effect if `miner` is set to `true`.
+
+[See this page for information on how to run a miner.](https://docs.blockstack.org/mining)
+
+Example:
+
+```toml
+mine_microblocks = true
+```
+
+#### prometheus_bind (optional)
+
+Address and port stacks-node should open for Prometheus metrics collection.
+
+Example:
+
+```toml
+prometheus_bind = "0.0.0.0:9153"
+```
+
+### Section: events_observer (optional)
+
+Contains options for watching events emitted by a local [stacks-blockchain-api](https://github.com/blockstack/stacks-blockchain-api) service.
+
+-> This section can be repeated multiple times.
+
+Example:
+
+```toml
+[[events_observer]]
+endpoint = "address-to-my-local.stacks-node-api.com:3700"
+retry_count = 255
+events_keys = ["*"]
+```
+
+#### endpoint
+
+Address and port to a stacks-node-api to watch for events.
+
+Example:
+
+```toml
+endpoint = "address-to-my-local.stacks-node-api.com:3700"
+```
+
+#### retry_count
+
+Number of times to retry before failing.
+
+Example:
+
+```toml
+retry_count = 255
+```
+
+#### events_keys
+
+Event keys for which to watch.
+
+Example:
+
+```toml
+events_keys = ["*"]
+```
+
+### Section: connection_options (optional)
+
+Specifies configuration options for others connecting to the stacks node.
+
+Example:
+
+```toml
+[connection_options]
+public_ip_address = "1.2.3.4:20444"
+download_interval = 10
+walk_interval = 30
+```
+
+#### public_ip_address
+
+The advertised public IP of this stacks-node.
+
+Example:
+
+```toml
+public_ip_address = "1.2.3.4:20444"
+```
+
+#### download_interval
+
+Time (in seconds) between attempts to download blocks.
+
+Example:
+
+```toml
+download_interval = 10
+```
+
+#### walk_interval
+
+Time (in seconds) between attempts to walk the Merkle tree.
+
+Example:
+
+```toml
+walk_interval = 30
+```
+
+### Section: burnchain
+
+Example:
+
+```toml
+[burnchain]
+chain = "bitcoin"
+mode = "krypton"
+peer_host = "bitcoind.blockstack.org"
+rpc_port = 18443
+peer_port = 18444
+```
+
+This section contains configuration options pertaining to the blockchain the stacks-node binds to on the backend for proof-of-transfer (BTC).
+
+#### chain
+
+The blockchain stacks-node binds to on the backend for proof-of-transfer.
+
+Example:
+
+```toml
+chain = "bitcoin"
+```
+
+#### mode
+
+The profile of which to run stacks-node.
```

Comment:
"profile", "mode" and sub commands are used to describe the same thing, however, it is not clear to me what the thing is.

Reply:
What are the valid values?

Reply:
Valid values can be found [here](https://github.com/blockstack/stacks-blockchain/blob/master/testnet/stacks-node/src/config.rs#L431), I'll include it in the description.

I said "profile" just to avoid re-using the word "mode" in the definition. I'm assuming this is a kind of feature toggle, and the subcommands effectively select the `mode` behind the scenes in the config they load in code.